### PR TITLE
MULE-19537: App get stuck when is trying to read the error payload in the error handler after a failing http request

### PR DESCRIPTION
**Pull request by rbourbonnavarro**

### CODE REVIEW

File: src/main/java/org/mule/service/scheduler/internal/threads/DedicatedSchedulerThreadPools.java [hunk 1]
```diff
@@ -108,6 +108,8 @@ protected void doStart(boolean preStartThreads) throws MuleException {
                                new SchedulerThreadFactory(computationGroup),
                                byCallerThreadGroupPolicy.apply(computationGroup.getName()));
 
+    preserveThreadOnErrorGroups = new HashSet<>(asList(ioGroup, customWaitGroup, customCallerRunsAnsWaitGroup));
```
Comment (elrodro83) [inline src/main/java/org/mule/service/scheduler/internal/threads/DedicatedSchedulerThreadPools.java:111]: in `createThreadGroupPolicy` this same set is already created. Extract and reuse.

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: src/main/java/org/mule/service/scheduler/internal/threads/SchedulerThreadPools.java [hunk 2]
```diff
@@ -157,6 +157,7 @@ public static Builder builder(String name, SchedulerPoolsConfig config) {
   protected final ThreadGroup customWaitGroup;
   protected final ThreadGroup customCallerRunsGroup;
   protected final ThreadGroup customCallerRunsAnsWaitGroup;
+  protected Set<ThreadGroup> waitGroups;
```
Comment (elrodro83) [inline src/main/java/org/mule/service/scheduler/internal/threads/SchedulerThreadPools.java:160]: move this field to the subclasses where this is set, add an abstract getter to use in this class

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: src/main/java/org/mule/service/scheduler/internal/threads/DedicatedSchedulerThreadPools.java (hunk 1 above)
Comment (rbourbonnavarro) [reply src/main/java/org/mule/service/scheduler/internal/threads/DedicatedSchedulerThreadPools.java]: Done.

File: src/main/java/org/mule/service/scheduler/internal/DefaultSchedulerService.java [hunk 3]
```diff
@@ -250,6 +252,23 @@ public boolean isCurrentThreadForCpuWork(SchedulerPoolsConfigFactory poolsConfig
     }
   }
 
+  @Override
+  public boolean isCurrentThreadInWaitGroup() {
+    return Boolean.TRUE.equals(waitGroupCache.get(currentThread()));
```
Comment (fsgonz) [inline src/main/java/org/mule/service/scheduler/internal/DefaultSchedulerService.java:257]: static import.

**Review by fsgonz — CHANGES_REQUESTED** [review]

**Review by fsgonz — APPROVED** [review]

File: src/main/java/org/mule/service/scheduler/internal/threads/SchedulerThreadPools.java (hunk 2 above)
Comment (rbourbonnavarro) [reply src/main/java/org/mule/service/scheduler/internal/threads/SchedulerThreadPools.java]: Done.

File: src/main/java/org/mule/service/scheduler/internal/DefaultSchedulerService.java (hunk 3 above)
Comment (rbourbonnavarro) [reply src/main/java/org/mule/service/scheduler/internal/DefaultSchedulerService.java]: Done.

**Review by fsgonz — APPROVED** [review]

**Pull request merged**